stand/lua/config.lua

```diff
@@ -578,8 +578,35 @@ function config.readConf(file, loaded_files)
 
 	loaded_files[file] = true
 
+	-- If product_vars is set, it must be a list of environment variable names
+	-- to walk through to guess the product.
+	--
+	-- And if a product can be guessed, prepend:
+	-- - /boot/PRODUCT.conf to loader_conf_files
+	-- - and /boot/loader.conf.d/PRODUCT/ to loader_conf_dirs
+	local product_vars = getEnv("product_vars")
+	local product
+	if product_vars ~= nil then
+		for var in product_vars:gmatch("%S+") do
+			product = getEnv(var)
+			if product ~= nil then
+				break
+			end
+		end
+	end
+
 	-- Going to process "loader_conf_files" extra-files
 	local loader_conf_files = getEnv("loader_conf_files")
+	if product ~= nil then
+		local product_conf_file = string.format("/boot/%s.conf", product)
+		if lfs.attributes(product_conf_file, "mode") == "file" then
+			if loader_conf_files == nil then
+				loader_conf_files = product_conf_file
+			else
+				loader_conf_files = product_conf_file .. " " .. loader_conf_files
+			end
+		end
+	end
```

Comment:
I like this, but think it would be a little better if you had the "for vars" loop around basically this entire section so you could do that for multiple variables that match. I might want to have certain variables set for all of my Dell servers, regardless of model, as well as overrides for a specific motherboard. If I set product_vars to "smbios.planar.maker smbios.planar.product" then I could do just that if the loop were around this whole thing.